app/scripts/components/exploration/data-utils-no-faux-module.ts

```diff
@@ -174,35 +174,27 @@ export function resolveRenderParams(
   datasetSourceParams: Record<string, any> | undefined,
   queryDataRenders: Record<string, any> | undefined
 ): Record<string, any> | undefined {
-  let renderParams: Record<string, any> | undefined;
   // Start with sourceParams from the dataset.
+  // Return the source param as it is if exists
   if (hasValidSourceParams(datasetSourceParams)) {
-    renderParams = datasetSourceParams;
+    return datasetSourceParams;
   }
 
-  // Check for the dashboard render configuration in queryData if not defined.
-  if (!renderParams && queryDataRenders?.dashboard) {
-    renderParams = queryDataRenders.dashboard;
+  // Check for the dashboard render configuration in queryData
+  if (!queryDataRenders) throw new Error ('No proper render parameter exists.');
```

Comment:
maybe different err messages for each spot to differentiate and find source of err right away - this line could be `No render data exists` ?